src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/ImportDocumentSnapshot.cs

```diff
@@ -3,61 +3,55 @@
 
 using System;
 using System.Diagnostics.CodeAnalysis;
-using System.IO;
 using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor;
 using Microsoft.AspNetCore.Razor.Language;
 using Microsoft.CodeAnalysis.Text;
 
 namespace Microsoft.CodeAnalysis.Razor.ProjectSystem;
 
-internal class ImportDocumentSnapshot : IDocumentSnapshot
+internal class ImportDocumentSnapshot(IProjectSnapshot project, RazorProjectItem item) : IDocumentSnapshot
 {
-    // The default import file does not have a kind or paths.
-    public string? FileKind => null;
-    public string? FilePath => null;
-    public string? TargetPath => null;
+    private static readonly Task<VersionStamp> s_versionTask = Task.FromResult(VersionStamp.Default);
 
-    public bool SupportsOutput => false;
-    public IProjectSnapshot Project => _project;
+    public IProjectSnapshot Project { get; } = project;
 
-    private readonly IProjectSnapshot _project;
-    private readonly RazorProjectItem _importItem;
+    private readonly RazorProjectItem _importItem = item;
     private SourceText? _sourceText;
-    private readonly VersionStamp _version;
 
-    public ImportDocumentSnapshot(IProjectSnapshot project, RazorProjectItem item)
-    {
-        _project = project;
-        _importItem = item;
-        _version = VersionStamp.Default;
-    }
+    // The default import file does not have a kind or paths.
+    public string? FileKind => null;
+    public string? FilePath => null;
+    public string? TargetPath => null;
 
     public int Version => 1;
 
-    public async Task<SourceText> GetTextAsync()
+    public Task<SourceText> GetTextAsync()
     {
-        using (var stream = _importItem.Read())
-        using (var reader = new StreamReader(stream))
+        return _sourceText is SourceText sourceText
+            ? Task.FromResult(sourceText)
+            : GetTextCoreAsync();
+
+        Task<SourceText> GetTextCoreAsync()
         {
-            var content = await reader.ReadToEndAsync().ConfigureAwait(false);
-            _sourceText = SourceText.From(content);
-        }
+            using var stream = _importItem.Read();
+            var sourceText = SourceText.From(stream);
 
-        return _sourceText;
+            var result = _sourceText ??= InterlockedOperations.Initialize(ref _sourceText, sourceText);
+            return Task.FromResult(result);
+        }
     }
 
-    public Task<RazorCodeDocument> GetGeneratedOutputAsync(bool _)
+    public Task<RazorCodeDocument> GetGeneratedOutputAsync(bool forceDesignTimeGeneratedOutput)
         => throw new NotSupportedException();
 
     public Task<VersionStamp> GetTextVersionAsync()
-    {
-        return Task.FromResult(_version);
-    }
+        => s_versionTask;
 
     public bool TryGetText([NotNullWhen(true)] out SourceText? result)
     {
-        if (_sourceText is { } sourceText)
+        if (_sourceText is SourceText sourceText)
```

Comment:
This is a surprising change. I thought using `{ }` was more the norm. No complaints, just curious for your thoughts.

Alternatively, this method could just be:

```
result = _sourceText;
return result is not null;
```

Reply:
> This is a surprising change. I thought using { } was more the norm. No complaints, just curious for your thoughts.

To be fair, this wasn't an isolated change. I actually deleted and rewrote the method, though the diff certainly makes it look as if I'd only changed `{ }` to `SourceText`. :smile:

I don't see the benefit of writing `is { } x` when the type is sufficiently short. I'm perfectly happy using `{ }` and have added many instances of that to this code base, However, it doesn't add a lot of benefit other than brevity and type-inference. So, if the type is short and non-generic, I might not always reach for the shift key to type `{ }`. Also, using a normal type test can improve readability when the expression to the left is complex. I guess my position is, when it comes to `is { } x`, I'm less a dogmatist and more of a pragmatist. I try to write code that is both succinct and readable, and avoid sacrificing one for the other.

**TL;DR! Anecdote follows!**

After putting in so much time on the C# language team working on pattern matching in the C# 7.x era, I still see `is { } x` as unfortunate fallout from the pattern matching design. In those days of yore, we debated whether empty property patterns should even be allowed because they don't actually match any properties. It was sort of a happy accident that `is { } x` is a succinct way to test that an expression is non-null reference type _and_ assign it to a type-inferred non-nullable variable. (It doesn't work for nullable value types BTW. So, `is { } x` can't be used universally.)

We did spend design effort looking for a better way to write a pattern that achieved the same goal. `is var x` was the most naturally syntax, but it already had a meaning. It does produce a type-inferred variable but doesn't perform a null test. (In fact, `is var x` [is kinda useless on its own](https://sharplab.io/#v2:EYLgtghglgdgPgAQEwEYCwAoBBmABM3AYVwG9NcL88EAWXAWQAoEUAGAflwGcBKU8yoKgAzXIy64oElq24o+LAJyMeAbgGCKIsRKm4AbhABO3JApTK1G3AF9MNoA)).  We thought about making `is not null x` work but that just looked plain weird to us, and it wasn't clear from the syntax that `x` would be type-inferred. Also, it wasn't as succinct. So, we left `is { } x` as a small compromise in the design. I’m at peace with that decision, but it’s also a small reminder that the design isn’t perfect. :smile:

**Related Aside!**

I really dislike that our `.editorconfig` enforces `var` so strictly. To me, it's especially silly with primitive types, where the language already provides succinct keywords. I’d love to see us dial our `.editorconfig` to not warn on primitive types. That way, I can write the plain and obvious `for (int i = 0; ...)` rather than having to always write `for (var i = 0; ...)`. Seems kinda arbitrary.

Reply:
> I'm less a dogmatist and more of a pragmatist

I know this about you, and find it very funny that the diff when viewed in isolation sends the exact opposite message 😁

I've always thought `if (x is not null y)` would be an okay syntax, but I guess 5 words separated with nothing but spaces is a little out of place in C#, other than linq. I definitely haven't thought about it for too long.

Reply:
> I really dislike that our .editorconfig enforces var so strictly

We accept PRs? :)

Reply:
> I've always thought `if (x is not null y)` would be an okay syntax, but I guess 5 words separated with nothing but spaces is a little out of place in C#, other than linq. I definitely haven't thought about it for too long

Yeah, it's an _OK_ syntax, but the double negative made it less ideal to us. (FWIW, the C# team had always had a bit of an aversion to VB 8.0's `x IsNot Nothing`.) Ultimately, we concluded that it wasn't worth replacing `is { } x` with `is not null x`. The first form was also an easier starting point if you wanted to add property matches later. c'est la vie!